src/humancellatlas/data/metadata/api.py

```diff
@@ -773,15 +814,19 @@ def from_json_vx(core_cls: Type[E], **kwargs) -> MutableMapping[UUID4, E]:
         self.entities = {**self.projects, **self.biomaterials, **self.processes, **self.protocols, **self.files}
 
         links = metadata_files['links.json']['links']
-        self.links = list(chain.from_iterable(map(Link.from_json, links)))
+        schema_version = tuple(int(x) for x in metadata_files['links.json']['schema_version'].split('.'))
+        self.links = []
+        for link in links:
+            self.links.extend(Link.from_json(link, schema_version))
```

Comment:
You can still use `chain.from_iterable`. Only the `map` should be replaced with a generator expression.